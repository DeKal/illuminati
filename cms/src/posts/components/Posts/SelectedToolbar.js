import React from 'react'
import PropTypes from 'prop-types'
import clsx from 'clsx'
import Toolbar from '@material-ui/core/Toolbar'
import Typography from '@material-ui/core/Typography'
import IconButton from '@material-ui/core/IconButton'
import Tooltip from '@material-ui/core/Tooltip'
import DeleteIcon from '@material-ui/icons/Delete'

const SelectedToolbar = ({ classes, numSelected }) => (
  <Toolbar
    className={clsx(classes.root, {
      [classes.highlight]: numSelected
    })}
  >
    {numSelected ? (
      <Typography className={classes.title} color="inherit" variant="subtitle1">
        {numSelected} selected
      </Typography>
    ) : (
      <Typography className={classes.title} variant="h6">
        All Posts
      </Typography>
    )}

    {numSelected ? (
      <Tooltip title="Delete">
        <IconButton aria-label="delete">
          <DeleteIcon />
        </IconButton>
      </Tooltip>
    ) : null}
  </Toolbar>
)

SelectedToolbar.propTypes = {
  numSelected: PropTypes.number.isRequired,
  classes: PropTypes.shape({
    root: PropTypes.object,
    title: PropTypes.object,
    highlight: PropTypes.object
  })
}

export default SelectedToolbar
