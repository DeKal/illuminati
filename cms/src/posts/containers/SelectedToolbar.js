import React from 'react'

import { lighten, makeStyles } from '@material-ui/core/styles'
import SelectedToolbar from 'posts/components/Posts/SelectedToolbar'

const useToolbarStyles = makeStyles(theme => ({
  root: {
    paddingLeft: theme.spacing(2),
    paddingRight: theme.spacing(1)
  },
  highlight: {
    color: theme.colors.primary,
    backgroundColor: lighten(theme.colors.primary, 0.85)
  },
  title: {
    flex: '1 1 100%'
  }
}))

const SelectedToolbarContainer = props => (
  <SelectedToolbar classes={useToolbarStyles()} {...props} />
)

export default SelectedToolbarContainer
