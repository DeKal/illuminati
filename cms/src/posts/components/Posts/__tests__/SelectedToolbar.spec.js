import * as React from 'react'
import { shallow } from 'enzyme'
import SelectedToolbar from '../SelectedToolbar'

const classes = {
  root: 'root',
  highlight: 'highlight',
  title: 'title'
}

describe('SelectedToolbar', () => {
  it('should render with 1 item selected', function() {
    const wrap = shallow(<SelectedToolbar classes={classes} numSelected={1} />)
    expect(wrap).toMatchSnapshot()
  })

  it('should render with no item selected', function() {
    const wrap = shallow(<SelectedToolbar classes={classes} numSelected={0} />)
    expect(wrap).toMatchSnapshot()
  })
})
