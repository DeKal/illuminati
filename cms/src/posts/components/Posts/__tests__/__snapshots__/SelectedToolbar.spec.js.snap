// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedToolbar should render with 1 item selected 1`] = `
<WithStyles(ForwardRef(Toolbar))
  className="root highlight"
>
  <WithStyles(ForwardRef(Typography))
    className="title"
    color="inherit"
    variant="subtitle1"
  >
    1
     selected
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Tooltip))
    title="Delete"
  >
    <WithStyles(ForwardRef(IconButton))
      aria-label="delete"
    >
      <DeleteIcon />
    </WithStyles(ForwardRef(IconButton))>
  </WithStyles(ForwardRef(Tooltip))>
</WithStyles(ForwardRef(Toolbar))>
`;

exports[`SelectedToolbar should render with no item selected 1`] = `
<WithStyles(ForwardRef(Toolbar))
  className="root"
>
  <WithStyles(ForwardRef(Typography))
    className="title"
    variant="h6"
  >
    All Posts
  </WithStyles(ForwardRef(Typography))>
</WithStyles(ForwardRef(Toolbar))>
`;
